client.c

```diff
@@ -1283,7 +1283,7 @@ static void client_flush_hist_samples(FILE *f, int hist_coarseness, void *sample
 	int log_offset;
 	uint64_t i, j, nr_samples;
 	struct io_u_plat_entry *entry;
-	unsigned int *io_u_plat;
+	unsigned long *io_u_plat;
```

Comment:
This should probably be uint64_t as well, unsigned long is only 32-bit on 32-bit archs.